src/plus/gk/account/subscriptionService.ts

```diff
@@ -874,6 +874,9 @@ export class SubscriptionService implements Disposable {
 
 			Logger.error(ex, scope);
 			debugger;
+
+			// If we cannot check in, validate stored subscription
+			this.changeSubscription(this._subscription);
```

Comment:
Correct me if I'm wrong but, shouldn't you spread `this._subscription` in as a copy since otherwise, it will be mutated, and the `matches` boolean in the function would always be `true`, preventing the subscription from properly being updated?

Reply:
I thought [this commit](https://github.com/gitkraken/vscode-gitlens/commit/faea28a390df2592f822cd7a0a211b299af486ee#diff-e821a438849eca64d4d0dc885d4fd3c752fcb059b657c75bbf3fe53b6ae939f3) would prevent just that.

Besides, I have only seen usages with the spread operator when overwriting values:
![Captura de pantalla 2024-07-26 a las 11 02 39](https://github.com/user-attachments/assets/bd60245e-9879-48a2-a314-ee594f531b18)
![Captura de pantalla 2024-07-26 a las 11 02 54](https://github.com/user-attachments/assets/2f124bd6-10c7-4ab7-8539-3e799aedc76b)
![Captura de pantalla 2024-07-26 a las 11 03 08](https://github.com/user-attachments/assets/07aaacc0-a6d3-4fb2-a376-90a757526c8f)
![Captura de pantalla 2024-07-26 a las 11 04 11](https://github.com/user-attachments/assets/e77b88d9-e62c-407a-98e6-3f8b8fc875f8)